torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -87,7 +87,7 @@ std::shared_ptr<NCCLComm> NCCLComm::create(
   comm->initialized_ = !comm->nonBlocking_;
```

Comment:
does this `!comm->nonBlocking` line not require HAS_COMM_NONBLOCKING feature as well?

Reply:
short answer: not required. This is a generic type (bool) that has no dependency on NCCL declaration.
(Otherwise, our code would look a bit more messy.)

If we generalize this usage, it can be viewed as a way to **unconditionally** create methods under `NCCLComm` even though older NCCL versions does not support it.

To give an example:
https://github.com/pytorch/pytorch/pull/145675/files#diff-c4e79e1dee3eea72b89d4471b91de8a7dae284e27bd5c32ed3e5296a561e4ee9R5257-R5268

It was due to the mem allocator creation prefers to use a function pointer `_ncclMemAlloc` unconditionally:
https://github.com/pytorch/pytorch/pull/145675/files#diff-c4e79e1dee3eea72b89d4471b91de8a7dae284e27bd5c32ed3e5296a561e4ee9R5285-R5288